### DISCUSSION TITLE
Добавить параметр base url

### DISCUSSION THREAD

**Discussion by 00svd00** [petrovvlad/freeLib#99] [Ideas]
Чтобы веб сервер корректно работал будучи спрятанным за реверс прокси

**Comment by petrovvlad** [comment]
Я могу ошибаться, но разве обязательно указывать base url для реверс прокси?

**Reply by 00svd00** [reply]
Не обязательно, но без этого получается достаточно некрасивая конструкция. Сейчас реверс прокси у меня заработал с таким конфигом(nginx)
location /http/ {
proxy_pass         http://127.0.0.1:8080/;
}
location ~ ^/opds/(.\*) {
proxy_pass         http://127.0.0.1:8080/opds/$1$is_args$args;
}
А хочется чтобы было что-то типа
location ~ ^/freelib/(.\*) {
proxy_pass         http://127.0.0.1:8080/$1$is_args$args;
}
Но для этого приложение должно понимать что оно живёт по пути $host/freelib и генерило сылки именно относительно этого пути

**Comment by gelato** [comment]
@00svd00 А можешь скинуть конфиг nginx? Я запарился уже мучаться opds заставить работать за проксей, у меня xml opds сервера как text отображается.

**Reply by 00svd00** [reply]
Ну прям щас в моменте так настроено
#FreeLib
location        /http/ {
        proxy_pass       http://127.0.0.1:8065/;
        rewrite ^/http/opds/$ /http/opds permanent;
}
location /author/ {
        rewrite ^/author/(.\*) /http/author/$1 last;
}
location /genres/ {
        rewrite ^/genres/(.\*) /http/genres/$1 last;
}
location ~ ^/opds/(.\*) {
        proxy_pass               http://127.0.0.1:8065/opds/$1$is_args$args;
        rewrite ^/opds/(.\*) /http/opds/$1 last;
}

На клиентах соотв. прописываем http://192.168.x.x/http/opds 
Работают вебморда и OPDS сервер(проверено в Alreader и CoolReader, в последнем не работает поиск)
По поводу того что xml отображается как text - ну у меня в хроме тоже так отображается. Это не проблема. Firefox в свою очередь качает файл - и это тоже не проблема, просто особенности браузеров.
Если что-то не работает, то это скорее всего кривой конфиг
А, да, забыл сказать, ещё кинул в кореневой каталог вебсервера home.png, писать для неё персональное правило мне показалось лютой дичью

**Reply by gelato** [reply]
Спасибо тебе огромное, теперь все фунциклирует как надо )

**Reply by accumail** [reply]
Имеется сайт на OpenServer на Apache (WS 2022LTSC), имеется freeLib 5.03 на 8888 порту.
Все работает: opds и web (включая отображение картинок).
Перевожу сайт на nginx 1.23, прописываю ваши настройки на нем в Nginx_1.23_vhost.conf для домена и получаю:
- да, web интерфейс freeLib стал наконец работать через https, но при этом возникли больше проблем, чем выгоды от использования https для web версии freeLib.
А именно:
1. в web версии freeLib полностью перестали отобржаться обложки,
2.  так на сайте в заголовках имеется общепринятая запись "Strict-Transport-Security  'max-age=63072000; includeSubDomains; preload'" то web интерфейс freeLib для http://домен:8888 более не доступен в Firefox (он остается доступным только по http://ip:8888).
3. общепрнятый плагин с миллионными установками Really Simple SSL (WordPress) вообще не понимает Nginx и пишет в диагностике безопасности сайта какой-то бред! 

P.S Так что выгода от использования обратного proxy сервера для искусственной поддержки https (которая не заложеная в самом продукте) - оказывается крайне сомнительной!
Мое личное мнение, никому и ничего не навязываю.

**Reply by accumail** [reply]
Автору форка добрый совет: если хотите, чтобы ваш продукт удовлетворял общепринятым нормам в безопасности и пользовался популярностью, то встройте в него нативную поддержку https! Вам решать...
